actors/organizer.go

```diff
@@ -144,168 +160,238 @@ func (o *Organizer) handlePublish(query message.Query) (message, channel []byte,
+			return nil, lib.ErrNotYetImplemented
 		default:
-			return nil, nil, lib.ErrInvalidAction
+			return nil, lib.ErrInvalidAction
 		}
 	default:
-		fmt.Printf("data[action] (%v) not recognized in handlepublish, generating default response ", data["action"])
-		return nil, nil, lib.ErrRequestDataInvalid
+		log.Printf("data[object] (%v) not recognized in handlepublish, generating default response ", data["object"])
+		return nil, lib.ErrRequestDataInvalid
 	}
 }
 
-/* handles the creation of a LAO */
-func (o *Organizer) handleCreateLAO(msg message.Message, canal string, query message.Query) (message, channel []byte, err error) {
+// handleCreateLAO is the function to handle a received message requesting a LAO Creation.
+// It is called by the function handlePublish.
+// The received message had the object field set to "lao" and action field to "create"
+// It will check for the validity of the received message, store the received message in the database, and store the new
+// LAO in the database.
+func (o *Organizer) handleCreateLAO(msg message.Message, canal string, query message.Query) (msgAndChannel []lib.MessageAndChannel, err_ error) {
 
 	if canal != "/root" {
-		return nil, nil, lib.ErrInvalidResource
+		log.Printf("Invalid channel. LAO create requests are valid only on the /root channel")
+		return nil, lib.ErrInvalidResource
 	}
 
 	data, errs := parser.ParseDataCreateLAO(msg.Data)
 	if errs != nil {
-		return nil, nil, lib.ErrInvalidResource
+		log.Printf("Could not parse received data into a create LAO structure")
+		return nil, lib.ErrInvalidResource
 	}
 
-	if !security.LAOIsValid(data, msg, true) {
-		return nil, nil, lib.ErrInvalidResource
-	}
-
-	errs = db.CreateMessage(msg, canal, o.database)
-	if errs != nil {
-		return nil, nil, err
+	if !security.LAOIsValid(data, true) {
+		log.Printf("Received data for LAO Creation not valid")
+		return nil, lib.ErrInvalidResource
 	}
 
 	lao := event.LAO{
-		ID:            data.ID,
+		ID:            string(data.ID),
 		Name:          data.Name,
 		Creation:      data.Creation,
-		OrganizerPKey: data.Organizer,
-		Witnesses:     data.Witnesses,
+		OrganizerPKey: string(data.Organizer),
+		Witnesses:     lib.ArrayArrayByteToArrayString(data.Witnesses),
 	}
+
 	errs = db.CreateChannel(lao, o.database)
 	if errs != nil {
-		return nil, nil, err
+		log.Printf("An error occured, could not create new channel in the database")
+		return nil, errs
+	}
+
+	errs = db.CreateMessage(msg, canal, o.database)
+	if errs != nil {
+		log.Printf("An error occured, could not store message to the database")
+		return nil, errs
 	}
 
-	return parser.ComposeBroadcastMessage(query), []byte(canal), nil
+	// as per hub.go, this msgAndChan will never be broadcast as Channel should be on root. We still create it in case this functionality could change in the future.
+	msgAndChan := []lib.MessageAndChannel{{
+		Message: parser.ComposeBroadcastMessage(query),
+		Channel: []byte(canal),
+	}}
```

Comment:
As this code is "standard and used elsewhere", I suggest we just remove it. (no need to compute extra stuff if not needed). And will make the code lighter